pkg/messaging/pubsub.go

```diff
@@ -5,6 +5,17 @@ package messaging
 
 import "context"
 
+type DeliveryPolicy uint8
+
+const (
+	// DeliverNewPolicy will only deliver new messages that are sent after the consumer is created.
+	// This is the default policy.
+	DeliverNewPolicy DeliveryPolicy = iota
+
+	// DeliverAllPolicy starts delivering messages from the very beginning of a stream.
+	DeliverAllPolicy
```

Comment:
This oddly resembles [JetStream deliver policy ](https://docs.nats.io/nats-concepts/jetstream/consumers#deliverpolicy. That's fine, but we need to check if there are other policies we'd like to support, such as deliver by start time, which requires additional context to the policy. We also need to check how it resonates with other brokers. Let's accept this one, but please open a ticket for potential improvements.

Reply:
https://github.com/absmach/magistrala/issues/9